File: Week2/Tuseday/armstrong/python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
def find_armstrong_numbers(numbers):
    armstrong_numbers = []
    for num in numbers:
        checkResults = armstrong_checker(num)
        if checkResults != None:
            armstrong_numbers.append(checkResults)
    return armstrong_numbers

def int_to_list(number):
    # using list comprehension to convert number to list of integers
    return [int(x) for x in str(number)]


def armstrong_checker(number):
    list = int_to_list(number)
    length = len(list)
    armstrong = 0
    for num in list:
        armstrong += num**length
    return armstrong if armstrong == number else None



# print(find_armstrong_numbers([10, 120, 3, 153]))

File: Week2/Tuseday/armstrong/js/armstrongNumbers.js
// How can you make this more scalable and reusable later?

const findArmstrongNumbers = (array) => {

  let armstrongNumbers = []
  for(num in array) {
    let checkResults = armstrongCheck(array[num])
    if(checkResults != null) {
      armstrongNumbers.push(checkResults)
    }
  } return armstrongNumbers
};

const intToArray = (int) => {
  return String(int).split('')
}

const armstrongCheck = (num) => {
  
  let array = intToArray(num)
  let length = array.length
  let armstrong = 0
  for(int in array){
    armstrong += Number(array[int])**length
  }
  return (armstrong == num ? armstrong:null)
}

console.log(findArmstrongNumbers([10, 120, 3, 153]))

module.exports = {
  findArmstrongNumbers
}
